sources/niUI/src/VideoDecoderTheoraPlayer.cpp

```diff
@@ -182,16 +172,38 @@ class cVideoDecoderTheoraPlayer : public ni::cIUnknownImpl<ni::iVideoDecoder,ni:
 
   ///////////////////////////////////////////////
   virtual tBool __stdcall Update(tBool abUpdateTarget, tF32 afFrameTime) {
-    if (!mpTheoraVideoClip)
+    if (!mpTheoraVideoClip || !mpTargetTexture)
       return eTrue;
 
     if (afFrameTime > 0.0f && !mbPaused) {
       mpTheoraVideoClip->update(afFrameTime);
       mbTargetDirty = eTrue;
+
+      WeakPtr<cVideoDecoderTheoraPlayer> _this = this;
+      mpExecutor->Execute(ni::Runnable([_this]() {
+        QPtr<cVideoDecoderTheoraPlayer> player(_this);
+        if (!player.IsOK()) return eTrue;
+        auto clip = player->mpTheoraVideoClip;
+        if (!clip) return eTrue;
+
+        if (clip->mSeekFrame >= 0) {
+          clip->doSeek();
+        }
+        clip->decodedAudioCheck();
+        clip->decodeNextFrame();
+        return eTrue;
+      }));
     }
 
     if (abUpdateTarget && mbTargetDirty && (mnLastUpdateFrameId != ni::GetLang()->GetFrameNumber())) {
-      _UpdateTarget();
+      auto f = mpTheoraVideoClip->getNextFrame();
```

Comment:
No auto pls. There's niCC.h, linter, etc... coming for these but generally the code isnt "auto safe" atm so its still banned until a proper cleanup/tooling is in place for it.